reports-api/src/reports_api/services/event_template.py

```diff
@@ -65,36 +70,125 @@ def import_events_template_new(cls, configuration_file):
             if selected_phase:
                 phase_result = selected_phase.update(phase_obj, commit=False)
             else:
-                phase_result = models.PhaseCode(**phase_obj).flush()
+                phase_result = PhaseCode(**phase_obj).flush()
             phase_result_copy = res.PhaseResponseSchema().dump(phase_result)
             phase_result_copy['events'] = []
-            parent_events = list(filter(lambda x, _phase_no=phase['no']: 'phase_no'
-                                        in x and x['phase_no'] == _phase_no and not x['parent_id'], event_dict))
-            existing_events = models.EventTemplate.find_by_phase_id(phase_result.id)
+            parent_events = copy.deepcopy(list(filter(lambda x, _phase_no=phase['no']: 'phase_no'
+                                                      in x and x['phase_no'] == _phase_no and not x['parent_id'],
+                                                      event_dict)))
+            existing_events = EventTemplate.find_by_phase_id(phase_result.id)
+            template_ids = list(map(lambda x: x.id, existing_events))
+            existing_outcomes = OutcomeTemplate.find_by_template_ids(template_ids)
+            outcome_ids = list(map(lambda x: x.id, existing_outcomes))
+            existing_actions = ActionTemplate.find_by_outcome_ids(outcome_ids)
             for event in parent_events:
                 event['phase_id'] = phase_result.id
                 event['start_at'] = str(event['start_at'])
                 event_result = cls._save_event_template(existing_events, event, phase_result.id)
-                (phase_result_copy['events']).append(res.EventTemplateResponseSchema().dump(event_result))
-                child_events = list(filter(lambda x, _parent_id=event['no']: 'parent_id'
-                                           in x and x['parent_id'] == _parent_id, event_dict))
+                child_events = copy.deepcopy(list(filter(lambda x, _parent_id=event['no']: 'parent_id'
+                                                         in x and x['parent_id'] == _parent_id, event_dict)))
+                outcome_dict = outcome_dict.replace({'event_template_id': event["no"]},
+                                                    {'event_template_id': event_result.id}, regex=True)
+                outcome_results = cls._handle_outcomes(outcome_dict, existing_outcomes,
+                                                       existing_actions, action_dict, event)
+                event_result_copy = res.EventTemplateResponseSchema().dump(event_result)
+                event_result_copy = res.EventTemplateResponseSchema().dump(event_result_copy)

```

Comment:
```python
    event_result_copy = res.EventTemplateResponseSchema().dump(event_result)
    event_result_copy = res.EventTemplateResponseSchema().dump(event_result_copy)
``` 
We are taking a response of `Schema.dump()` and passing it to `Schema.dump()` of same schema again. Did you mean to `load`?